scipy/optimize/_particleswarmoptimization.py

```diff
@@ -334,93 +398,103 @@ def _test(self):
-	        if (np.size(self.velocity_bounds, 1) != 2 or not 
-	            np.all(np.isfinite(self.velocity_bounds))):
-	            raise ValueError('velocity_bounds should be a sequence'
-	            	             ' containing real valued (min, max) pairs for'
-	            	             ' each value in x')
+            if np.size(self.velocity_bounds, 1) != 2 or not np.all(np.isfinite(self.velocity_bounds)):
+                raise ValueError('velocity_bounds should be a sequence'
+                                 ' containing real valued (min, max) pairs for'
+                                 ' each value in x')
 
         # test mutation
         if (self.mutation >= 1) or (self.mutation < 0):
             raise ValueError('The mutation value must be in the'
                              ' interval [0, 1)')
 
-    def _particle_initialization(self):
-        """ Initialization of particle values """
-
-        particle_vector = self.swarm
+        # test seed
+        if (type(self.seed) is not int) or (type(self.seed) is not None):
+            raise ValueError('seed must be int')
 
-        for part in particle_vector:
+    def _initialization(self):
+        """ Initialization of the swarm particles """
+        for i in range(self.n_particles):
+            self._particle_initialization(i)
+            if self.particles[i].bestof <= self.bestgof:
+                self.bestgpos = self.particles[i].bestpos
+                self.bestgof = self.particles[i].bestof
 
-            # Random position in the swarm
-            part.position = self._random_generation(self.initial_bounds)
+        self.fevals += self.n_particles
+        self.it += 1
 
-            # Velocity initialization
-            part.velocity = np.zeros(self.num_param)
-
-            # Evaluation
-            part.function_eval = self.func(part.position)
-            self.it += 1
-
-            # Individual best update
-            part.best_position = part.position
-            part.best_function_eval = part.function_eval
-
-            # Global best update
-            if part.best_function_eval < self.best_global_function_eval:
-                self.best_global_function_eval = part.best_function_eval
-                self.best_global_position = part.best_position
-
-    def _particle_update(self):
+    def _particle_initialization(self, i):
+        """ Initialization of particle values """
+        init_position = self._particle_random_generation(self.bounds)
+        self.particles[i] = Particle(position=init_position,
+                                     velocity=0,
+                                     bestpos=init_position,
+                                     bestof=self.func(init_position),
+                                     func=self.func)
+
+    def _update(self):
+        """ Update of the swarm particles """
+        for i in range(self.n_particles):
+            self._particle_update(i)
+            if self.particles[i].bestof <= self.bestgof:
```

Comment:
Is this parallelisable? If so, then we should use a `workers` keyword to enable parallelisation (see differential_evolution).